### PR TITLE
EP-140: Uploading Codacy report

### PR DESCRIPTION
**Pull request by aslotte**
## JIRA
EP-140

## Description
This is a PoC regarding a code coverage tool called Codacy.
Once Codacy has been evaluated against other tools such as Sonar, it will be decided if this change is persistent or will be reverted.

### CODE REVIEW

File: .github/workflows/code_quality.yml [hunk 1]
```diff
@@ -10,4 +10,19 @@ jobs:
     uses: tetrascience/ts-data-insights-ci-cd-common/.github/workflows/python_code_quality.yml@main
     with:
       python-version: 3.7
-      run_tests: true
\ No newline at end of file
+      run_tests: true
+
+  upload_code_coverage:
```
Comment (VladimirShteyn) [inline .github/workflows/code_quality.yml:15]: If the PoC is approved, will this be moved to `ts-data-insights-ci-cd-common`?

File: .github/workflows/code_quality.yml [hunk 2]
```diff
@@ -10,4 +10,19 @@ jobs:
...
+    runs-on: ubuntu-latest
+    needs: reusable_workflow
+    steps:
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: codecoverage
```
Comment (VladimirShteyn) [inline .github/workflows/code_quality.yml:23]: [vlad edification] How does `actions/download-artifact@v3` "know" which file to download?

File: .github/workflows/code_quality.yml (hunk 1 above)
Comment (aslotte) [reply .github/workflows/code_quality.yml:15]: I think that would make sense yes and enable code coverage on all DI repos by default

File: .github/workflows/code_quality.yml (hunk 2 above)
Comment (aslotte) [reply .github/workflows/code_quality.yml:23]: By name. The name of the uploaded artifact is `codecoverage` in the `ts-data-insight-ci-cd-common` repository's code_quality workflow

Comment (VladimirShteyn) [reply .github/workflows/code_quality.yml:23]: Ah ok, it's the name [here](https://github.com/tetrascience/ts-data-insights-ci-cd-common/blob/main/.github/workflows/python_code_quality.yml#L102). I recommend adding a comment stating where `codecoverage` comes from.

**Review by VladimirShteyn — APPROVED** [review]

**Pull request merged**